### DISCUSSION TITLE
Filter loose and exact on same field using different GET parameters

### DISCUSSION THREAD

**Discussion by akulmehta** [spatie/laravel-query-builder#546] [Q&A] [answered]
I have a usecase in which I want to let the user be able to filter on a column/field either loose or exact. How can I write the `allowedFilters` parameters to cover this scenario? 

Example:

If the get request is: 
/posts?filter[title]=Lorem
It should do a regular allowedFilters i.e. not exact 

But if the request is:
/posts?filter[title_exact]=Lorem
It should list the post with the title exactly "Lorem"

**Comment by dominikb** [comment, marked as answer]
Hi @akulmehta,

you almost described the solution yourself. Translated into a `QueryBuilder` it can look like this:

```php
QueryBuilder::for(Post::class)
    ->allowedFilters([
        AllowedFilter::partial('title'),
        AllowedFilter::exact('title_exact', 'title'),
    ])
    ->get();
```

**Reply by akulmehta** [reply]
Hi @dominikb, sorry I am quite new to this.

But would this make QueryBuilder look specifically at the `title` column for both partial and exact cases?

Alternatively, if I do a partial case for say a request with quotes then will QueryBuilder turn that specific case into an exact.
eg. for
/posts?filter[title]=Lorem
gives partial but
/posts?filter[title]="Lorem"
gives exact?

**Reply by akulmehta** [reply]
Nevermind. I found my answer.
I will have to use Filter-aliases for the exact as shown https://spatie.be/docs/laravel-query-builder/v3/features/filtering#filter-aliases

**Reply by dominikb** [reply]
Forgot to add the property alias in my initial reply - sorry 😄 

Using a custom filter you could implement the query logic with quotes as well. But this could lead to edge cases with quotation marks in the actual title etc.
